### ISSUE TITLE
Fails to run when not connected to the Internet

### ISSUE THREAD

**Issue by probonopd** [ShuttleBrowser/Shuttle#50]
Fails to run when not connected to the Internet:

```
A JavaScript error occurred in the main process
Uncaught Exception:
Error: ENOENT: no such file or directory, open '/home/travis/.config/shuttle/settings.json'
    at Object.openSync (fs.js:436:3)
    at Object.module.(anonymous function) [as openSync] (ELECTRON_ASAR.js:160:31)
    at Object.writeFileSync (fs.js:1168:35)
    at Object.<anonymous> (/run/firejail/appimage/.appimage-10295/resources/app.asar/app/modules/files.js:19:6)
    at Object.<anonymous> (/run/firejail/appimage/.appimage-10295/resources/app.asar/app/modules/files.js:154:3)
    at Module._compile (internal/modules/cjs/loader.js:693:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:704:10)
    at Module.load (internal/modules/cjs/loader.js:602:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:541:12)
    at Function.Module._load (internal/modules/cjs/loader.js:533:3)
Fontconfig warning: "/etc/fonts/fonts.conf", line 86: unknown element "blank"
Xlib:  extension "RANDR" missing on display ":99".
/proc/self/exe: /lib/x86_64-linux-gnu/libdbus-1.so.3: no version information available (required by /proc/self/exe)
Checking for update
ATTENTION: default value of option force_s3tc_enable overridden by environment.
checking for update
Generated new staging user ID: b3fc8bc3-b1f4-58cb-a6f7-f5aaee1e3dac
Error: Error: net::ERR_INTERNET_DISCONNECTED
    at GenericProvider.getLatestVersion (/run/firejail/appimage/.appimage-10295/resources/app.asar/node_modules/electron-updater/src/providers/GenericProvider.ts:18:25)
    at /run/firejail/appimage/.appimage-10295/resources/app.asar/node_modules/electron-updater/src/AppUpdater.ts:347:26
    at Generator.next (<anonymous>)
    at runCallback (timers.js:694:18)
    at tryOnImmediate (timers.js:665:5)
    at processImmediate (timers.js:647:5)
From previous event:
    at AppImageUpdater.getUpdateInfoAndProvider (/run/firejail/appimage/.appimage-10295/resources/app.asar/node_modules/electron-updater/src/AppUpdater.ts:336:43)
    at /run/firejail/appimage/.appimage-10295/resources/app.asar/node_modules/electron-updater/src/AppUpdater.ts:363:31
    at Generator.next (<anonymous>)
From previous event:
    at AppImageUpdater.doCheckForUpdates (/run/firejail/appimage/.appimage-10295/resources/app.asar/node_modules/electron-updater/src/AppUpdater.ts:360:34)
    at AppImageUpdater.checkForUpdates (/run/firejail/appimage/.appimage-10295/resources/app.asar/node_modules/electron-updater/src/AppUpdater.ts:220:35)
    at App.app.on (/run/firejail/appimage/.appimage-10295/resources/app.asar/main/updater.js:16:15)
    at App.emit (events.js:187:15)
Error: net::ERR_INTERNET_DISCONNECTED
    at GenericProvider.getLatestVersion (/run/firejail/appimage/.appimage-10295/resources/app.asar/node_modules/electron-updater/src/providers/GenericProvider.ts:18:25)
    at /run/firejail/appimage/.appimage-10295/resources/app.asar/node_modules/electron-updater/src/AppUpdater.ts:347:26
    at Generator.next (<anonymous>)
    at runCallback (timers.js:694:18)
    at tryOnImmediate (timers.js:665:5)
    at processImmediate (timers.js:647:5)
From previous event:
    at AppImageUpdater.getUpdateInfoAndProvider (/run/firejail/appimage/.appimage-10295/resources/app.asar/node_modules/electron-updater/src/AppUpdater.ts:336:43)
    at /run/firejail/appimage/.appimage-10295/resources/app.asar/node_modules/electron-updater/src/AppUpdater.ts:363:31
    at Generator.next (<anonymous>)
From previous event:
    at AppImageUpdater.doCheckForUpdates (/run/firejail/appimage/.appimage-10295/resources/app.asar/node_modules/electron-updater/src/AppUpdater.ts:360:34)
    at AppImageUpdater.checkForUpdates (/run/firejail/appimage/.appimage-10295/resources/app.asar/node_modules/electron-updater/src/AppUpdater.ts:220:35)
    at App.app.on (/run/firejail/appimage/.appimage-10295/resources/app.asar/main/updater.js:16:15)
    at App.emit (events.js:187:15)
Unhandled rejection Error: net::ERR_INTERNET_DISCONNECTED
    at GenericProvider.getLatestVersion (/run/firejail/appimage/.appimage-10295/resources/app.asar/node_modules/electron-updater/src/providers/GenericProvider.ts:18:25)
    at /run/firejail/appimage/.appimage-10295/resources/app.asar/node_modules/electron-updater/src/AppUpdater.ts:347:26
    at Generator.next (<anonymous>)
    at runCallback (timers.js:694:18)
    at tryOnImmediate (timers.js:665:5)
    at processImmediate (timers.js:647:5)
From previous event:
    at AppImageUpdater.getUpdateInfoAndProvider (/run/firejail/appimage/.appimage-10295/resources/app.asar/node_modules/electron-updater/src/AppUpdater.ts:336:43)
    at /run/firejail/appimage/.appimage-10295/resources/app.asar/node_modules/electron-updater/src/AppUpdater.ts:363:31
    at Generator.next (<anonymous>)
From previous event:
    at AppImageUpdater.doCheckForUpdates (/run/firejail/appimage/.appimage-10295/resources/app.asar/node_modules/electron-updater/src/AppUpdater.ts:360:34)
    at AppImageUpdater.checkForUpdates (/run/firejail/appimage/.appimage-10295/resources/app.asar/node_modules/electron-updater/src/AppUpdater.ts:220:35)
    at App.app.on (/run/firejail/appimage/.appimage-10295/resources/app.asar/main/updater.js:16:15)
    at App.emit (events.js:187:15)
```

Reference:
https://github.com/AppImage/appimage.github.io/pull/1128

**Comment by robjullian** [comment]
this is just an error generated by the auto-updater but the real error is : 
`
Error: ENOENT: no such file or directory, open '/home/travis/.config/shuttle/settings.json'
    at Object.openSync (fs.js:436:3)
    at Object.module.(anonymous function) [as openSync] (ELECTRON_ASAR.js:160:31)
    at Object.writeFileSync (fs.js:1168:35)
    at Object.<anonymous> (/run/firejail/appimage/.appimage-10295/resources/app.asar/app/modules/files.js:19:6)
    at Object.<anonymous> (/run/firejail/appimage/.appimage-10295/resources/app.asar/app/modules/files.js:154:3)
    at Module._compile (internal/modules/cjs/loader.js:693:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:704:10)
    at Module.load (internal/modules/cjs/loader.js:602:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:541:12)
    at Function.Module._load (internal/modules/cjs/loader.js:533:3)
`
The settings.json file is not found, but its theoretically fixed in the latest version : https://github.com/ShuttleBrowser/Shuttle/releases/tag/3.0.6-beta.7

**Comment by robjullian** [comment]
Fixed in 3.0.10 https://github.com/ShuttleBrowser/Shuttle/releases/tag/v3.0.10

**Comment by probonopd** [comment]
Thanks, works.
